File: cf/b_261.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <array>
#include <algorithm>
#include <utility>
#include <map>
#include <queue>
#include <set>
#include <cmath>
#include <cstdio>
#include <cstring>
#include <functional>
#include <numeric>

#define ll long long
#define MOD 1000000007
#define INF 0x3f3f3f3f
#define INFLL 0x3f3f3f3f3f3f3f3f
#define MAXN 100000

using namespace std;

constexpr int sz = 200005;
ll b[sz];

int main()
{
	int n;
	cin >> n;
	ll maxb = -INFLL;
	ll minb = INFLL;
	for(int i = 0; i < n; i++)
	{
		cin >> b[i];
		maxb = max(b[i], maxb);
		minb = min(b[i], minb);
	}
	ll nmin = 0;
	ll nmax = 0;
	for(int i = 0; i < n; i++)
	{
		nmax += (b[i] == maxb);
		nmin += (b[i] == minb);
	}
	cout << maxb - minb << " " << nmax * nmin << endl;
	return 0;
}

File: cf/b_261.py
import sys

reader = sys.stdin.readline

def ri():
    return int(reader())

def rlst_i():
    return list(map(int, reader().split()))

def rstr():
    s = reader()
    return list(s[:len(s) - 1])

def solve():
    n, b = ri(), rlst_i()

    maxb, minb = -1e9, 1e9
    for beauty in b:
        maxb = max(maxb, beauty)
        minb = min(minb, beauty)
    
    nmax, nmin = 0, 0
    for beauty in b:
        if beauty == maxb:
            nmax += 1
        elif beauty == minb:
            nmin += 1

    max_diff = int(maxb - minb)
    max_diff_num_ways = int(nmax * (nmax - 1) // 2 if maxb == minb else nmax * nmin)

    print(str(max_diff) + " " + str(max_diff_num_ways))

if __name__ == '__main__':
    solve()
